File: Boot/Brasserie/src/main/java/soprajc/Brasserie/model/Snack.java
package soprajc.Brasserie.model;

import java.util.List;
import javax.persistence.DiscriminatorValue;
import javax.persistence.Entity;
import javax.persistence.ManyToMany;
import javax.persistence.PrimaryKeyJoinColumn;
import javax.persistence.Version;

@Entity
@DiscriminatorValue("Snack")

public class Snack extends Produit{

	@ManyToMany(mappedBy= "suggestions")
	private List<Biere> bieres;
	
	
	public Snack() {}
	
	public Snack(String nom, String description, double prix,  int stock) {
		super(nom, description, prix, stock);
	}
	
	public Snack(Integer id, String nom, String description, double prix,  int stock) {
		super(id, nom, description, prix, stock);
	}
	
	
	public Snack(String nom, String description, double prix,  int stock, List<Biere> bieres) {
		super(nom, description, prix, stock);
		this.bieres = bieres;
	}
	
	
	//Constructeur test
	public Snack(String nom, double prix) {
		super(nom, prix);
	}
	

	public List<Biere> getBieres() {
		return bieres;
	}

	public void setBieres(List<Biere> bieres) {
		this.bieres = bieres;
	}

		
	public List<Snack> suggestion(Biere biere){
		return null;
	}
}

File: Angular/Brasserie/src/app/model/snack.ts
import { Biere } from './biere';
import { Produit } from './produit';
export class Snack extends Produit {
  public constructor(
    id?: number | undefined,
    nom?: string | undefined,
    description?: string | undefined,
    prix?: number | undefined,
    stock?: number | undefined,
    points?: number | undefined,
    private _bieres?: Biere[] | undefined
  ) {
    super(id, nom, description, prix, stock, points);
  }

  public get bieres(): Biere[] | undefined {
    return this.bieres;
  }

  public set bieres(value: Biere[] | undefined) {
    this._bieres = value;
  }
}
